gateway/src/main/java/io/zeebe/gateway/impl/broker/BrokerRequestManager.java

```diff
@@ -115,37 +116,57 @@ private static boolean responseValidation(final DirectBuffer responseContent) {
       nodeIdProvider = determineBrokerNodeIdProvider(request);
     } catch (final PartitionNotFoundException e) {
       returnFuture.completeExceptionally(e);
+      GatewayMetrics.registerFailedRequest(request.getPartitionId(), request.getType());
       return;
     }
 
     final ActorFuture<DirectBuffer> responseFuture =
         sender.send(clientTransport, nodeIdProvider, request, requestTimeout);
+    final long startTime = System.currentTimeMillis();
 
     actor.runOnCompletion(
         responseFuture,
         (clientResponse, error) -> {
           try {
             if (error == null) {
               final BrokerResponse<T> response = request.getResponse(clientResponse);
-              handleResponse(response, returnFuture);
+
+              if (handleResponse(response, returnFuture)) {
+                final long elapsedTime = System.currentTimeMillis() - startTime;
+                GatewayMetrics.registerSuccessfulRequest(
+                    request.getPartitionId(), request.getType(), elapsedTime);
+                return;
+              }
             } else {
               returnFuture.completeExceptionally(error);
             }
           } catch (final RuntimeException e) {
             returnFuture.completeExceptionally(new ClientResponseException(e));
           }
+
+          GatewayMetrics.registerFailedRequest(request.getPartitionId(), request.getType());
         });
   }
 
-  private <T> void handleResponse(
+  /**
+   * Returns true if the request was successfully processed or false, it it wasn't. Rejections and
+   * RESOURCE_EXCEEDED errors are considered successful requests.
+   */
+  private <T> boolean handleResponse(
       final BrokerResponse<T> response, final CompletableFuture<BrokerResponse<T>> responseFuture) {
     try {
       if (response.isResponse()) {
         responseFuture.complete(response);
+        return true;
       } else if (response.isRejection()) {
         responseFuture.completeExceptionally(new BrokerRejectionException(response.getRejection()));
+        return true;
       } else if (response.isError()) {
         responseFuture.completeExceptionally(new BrokerErrorException(response.getError()));
+
+        if (response.getError().getCode() == ErrorCode.RESOURCE_EXHAUSTED) {
```

Comment:
I'm not sure if this should be considered as successful.

Reply:
Nicolas asked for it specifically. IIRC, the idea was that marking these as failed would show an increase in failures when a client sends too many requests, which isn't necessarily indicative of a problem in Zeebe. In this sense, including it would reduce the usefulness of have this metric. But @npepinpe, correct me if I'm wrong or misremembering

Reply:
The idea was that if we want to track the error rate, when stress testing RESOURCE_EXHAUSTED would show a system which is failing when it's behaving as expected.

Reply:
My main concern is that if there are too many resource exhausted errors during a stress test, we highly underestimate average latency because resource exhausted responses will have comparatively much lower latency.

Reply:
We can have both by ignoring them altogether. We don't count them as failures and we also don't register their latencies.